src/lib/memory/shard_manager.cpp

```diff
@@ -6,38 +6,30 @@ namespace homeobject {
 
 uint64_t ShardManager::max_shard_size() { return Gi; }
 
-static uint64_t get_current_timestamp() {
-    auto now = std::chrono::system_clock::now();
-    auto duration = std::chrono::duration_cast< std::chrono::milliseconds >(now.time_since_epoch());
-    auto timestamp = static_cast< uint64_t >(duration.count());
-    return timestamp;
-}
-
 ShardManager::Result< ShardInfo > MemoryHomeObject::_create_shard(pg_id pg_owner, uint64_t size_bytes) {
     auto lg = std::scoped_lock(_pg_lock, _shard_lock);
     auto pg_it = _pg_map.find(pg_owner);
     if (_pg_map.end() == pg_it) return folly::makeUnexpected(ShardError::UNKNOWN_PG);
 
     auto const now = get_current_timestamp();
-    auto& s_set = pg_it->second.second;
-    auto s_id = (((uint64_t)pg_owner) << 48) + s_set.size();
+    auto& shard_list = pg_it->second.shards;
+    auto s_id = (((uint64_t)pg_owner) << 48) + shard_list.size();
     auto info = ShardInfo{s_id, pg_owner, ShardInfo::State::OPEN, now, now, size_bytes, size_bytes, 0};
+    auto cinfo = std::make_shared<ShardInfo>(info);
```

Comment:
Can we combine the above two lines? No need for the temporary variable that I see.

Reply:
sure